Modules/Prospect/Entities/Prospect.php

```diff
@@ -38,12 +38,24 @@ public function getProspectDisplayName()
         return $this->organization_name ?? optional($this->client)->name ?? 'N/A';
     }
 
-    public function formattedAmount($amount)
+    public function getFormattedBudgetAttribute()
     {
-        $formattedAmount = preg_replace('/\B(?=(\d{2})+(?!\d))/', ',', substr($amount, 0, -3)) .
-                    ',' . substr($amount, -3);
+        $budget = (string) $this->budget;
+        $currency = $this->currency;
 
-        return $formattedAmount;
+        // if currency is less than one thousand
+        if (strlen($budget) <= 3) {
+            return $budget;
+        }
+
+        if ($currency == 'INR') {
+            $formattedBudget = preg_replace('/\B(?=(\d{2})+(?!\d))/', ',', substr($budget, 0, -3)) .
+                    ',' . substr($budget, -3);
```

Comment:
Wouldn't you prefer PHP's very own `NumberFormatter` class over this custom regex based formatting?
```suggestion
            $formatter = new \NumberFormatter('en_IN', \NumberFormatter::DECIMAL);
            $formattedBudget = $formatter->format($budget);
```